torch/distributed/_shard/sharded_tensor/_ops/tensor_ops.py

```diff
@@ -43,7 +43,7 @@ def tensor_device(types, args=(), kwargs=None, pg=None):
     if not isinstance(self_st, ShardedTensor):
         raise TypeError("input needs to be a ShardedTensor")
 
-    return self_st.local_shards()[0].tensor.device
+    return self_st.local_shards()[0].tensor.device if self_st.local_shards() else None
```

Comment:
This looks ok if no local shards return None, I'm wondering in the case of missing shards, do you want to return None, or if you want to return `torch.cuda.current_device()` instead?

Reply:
> I'm wondering in the case of missing shards

What do you mean by missing shards?

My assumption is that if local_shards() returns None, then there is not local shards for this ShardedTensor (even though this rank is still aware). IMO NONE is the correct device, but I'm open to other options